var/spack/repos/builtin/packages/libtheora/package.py

```diff
@@ -40,7 +67,60 @@ def autoreconf(self, spec, prefix):
         else:
             sh("./autogen.sh", "prefix={0}".format(prefix))
 
-    def configure_args(self):
-        args = []
-        args += self.enable_or_disable("doc")
-        return args
+
+class MSBuildBuilder(MSBuildBuilder):
+    def is_64bit(self):
+        return platform.machine().endswith("64")
+
+    def setup_build_environment(self, env):
+        spec = self.pkg.spec
+        env.set("SPACK_OGG_PREFIX", spec["libogg"].prefix)
+        # devenv is needed to convert ancient MSbuild project to modern
+        # msbuild project so MSBuild versions older than 2010 can build this
+        # project
+        devenv_path = os.path.join(self.pkg.compiler.vs_root, "Common7", "IDE")
+        env.prepend_path("PATH", devenv_path)
+
+    @property
+    def build_directory(self):
+        win_dir = os.path.join(super().build_directory, "win32")
+        vs_dir = "VS2008"
+        return os.path.join(win_dir, vs_dir)
+
+    @property
+    def sln_file(self):
+        if self.pkg.spec.satisfies("+static"):
+            f = "libtheora_static.sln"
+        else:
+            f = "libtheora_dynamic.sln"
+        return f
+
+    def msbuild_args(self):
+        return [self.define("Configuration", "Release"), self.sln_file]
+
+    def build(self, pkg, spec, prefix):
+        with working_dir(self.build_directory):
+            devenv = Executable("devenv")
+            devenv(self.sln_file, "/Upgrade")
+        super().build(pkg, spec, prefix)
+
+    def install(self, pkg, spec, prefix):
+        if not os.path.isdir(prefix.lib):
+            mkdirp(prefix.lib)
+        libs_to_install = glob.glob(
+            os.path.join(self.build_directory, "**", "*.lib"), recursive=True
+        )
+        for library in libs_to_install:
+            install(library, prefix.lib)
+        rename(
+            os.path.join(prefix.lib, "libtheora_static.lib"),
+            os.path.join(prefix.lib, "theora.lib"),
+        )
+        # The encoder and decoder libs are baked into the theora lib on Windows, so we spoof them
+        # here so libtheora is detectable by CMake and pkg-config and linkable by
+        # consuming projects
+        with working_dir(prefix.lib):
+            copy("theora.lib", "theoradec.lib")
+            copy("theora.lib", "theoraenc.lib")
+        with working_dir(pkg.stage.source_path):
+            install_tree("include", prefix.include)
```

Comment:
(question) is a `.dll` installed for `~static`? It wasn't clear to me looking through this function (i.e. I don't see a search for `.dll`).

Reply:
Very likely yes. This was an error on my part as I was too focused on what Paraview needs. I'll add the dlls to the install line.

Reply:
This has been done.